liquibase-core/src/test/groovy/liquibase/datatype/DataTypeFactoryTest.groovy

```diff
@@ -210,8 +210,50 @@ class DataTypeFactoryTest extends Specification {
+        "java.sql.Types.TIMESTAMP"                     | new PostgresDatabase() | "TIMESTAMP WITHOUT TIME ZONE"                  | TimestampType | false
+        "java.sql.Types.TIMESTAMP(6)"                  | new PostgresDatabase() | "TIMESTAMP(6) WITHOUT TIME ZONE"               | TimestampType | false
+        "java.sql.Types.TIMESTAMP_WITH_TIMEZONE"       | new PostgresDatabase() | "TIMESTAMP WITH TIME ZONE"                     | TimestampType | false
+        "java.sql.Types.TIMESTAMP_WITH_TIMEZONE(6)"    | new PostgresDatabase() | "TIMESTAMP(6) WITH TIME ZONE"                  | TimestampType | false
         "BINARY(16)"                                   | new H2Database()       | "BINARY(16)"                                   | BlobType      | false
+        "timestamp"                                    | new H2Database()       | "TIMESTAMP"                                    | TimestampType | false
+        "timestamp(6)"                                 | new H2Database()       | "TIMESTAMP(6)"                                 | TimestampType | false
+        "TIMESTAMP WITH TIMEZONE"                      | new H2Database()       | "TIMESTAMP WITH TIMEZONE"                      | TimestampType | false
+        "TIMESTAMP(6) WITH TIMEZONE"                   | new H2Database()       | "TIMESTAMP(6) WITH TIMEZONE"                   | TimestampType | false
+        "timestamp without timezone"                   | new H2Database()       | "TIMESTAMP"                                    | TimestampType | false
+        "timestamp(6) without timezone"                | new H2Database()       | "TIMESTAMP(6)"                                 | TimestampType | false
+        "timestamptz"                                  | new H2Database()       | "TIMESTAMP"                                    | TimestampType | false
+        "timestamptz(6)"                               | new H2Database()       | "TIMESTAMP(6)"                                 | TimestampType | false
+        "java.sql.Timestamp"                           | new H2Database()       | "TIMESTAMP"                                    | TimestampType | false
+        "java.sql.Timestamp(6)"                        | new H2Database()       | "TIMESTAMP(6)"                                 | TimestampType | false
+        "java.sql.Types.TIMESTAMP"                     | new H2Database()       | "TIMESTAMP"                                    | TimestampType | false
+        "java.sql.Types.TIMESTAMP(6)"                  | new H2Database()       | "TIMESTAMP(6)"                                 | TimestampType | false
+        "java.sql.Types.TIMESTAMP_WITH_TIMEZONE"       | new H2Database()       | "TIMESTAMP WITH TIMEZONE"                      | TimestampType | false
+        "java.sql.Types.TIMESTAMP_WITH_TIMEZONE(6)"    | new H2Database()       | "TIMESTAMP(6) WITH TIMEZONE"                   | TimestampType | false
```

Comment:
The same problem here.

Note that H2 supports `TIMESTAMP WITH TIME ZONE` only since 1.4.197, in older versions this data type was experimental only and was not really usable.

Reply:
Thanks, I wasn't aware of that change. I've pushed a new commit to correct the generated SQL.